exchange/utils.go

```diff
@@ -149,102 +147,110 @@ func (rs *requestSplitter) cleanOpenRTBRequests(ctx context.Context,
 		gdprPerms = rs.gdprPermsBuilder(auctionReq.TCF2Config, gdprRequestInfo)
 	}
 
-	// bidder level privacy policies
+	allowedBidderRequests = make([]BidderRequest, 0, len(allBidderRequests))
+
 	for _, bidderRequest := range allBidderRequests {
-		// fetchBids activity
-		scopedName := privacy.Component{Type: privacy.ComponentTypeBidder, Name: bidderRequest.BidderName.String()}
-		fetchBidsActivityAllowed := auctionReq.Activities.Allow(privacy.ActivityFetchBids, scopedName, privacy.NewRequestFromBidRequest(*req))
-		if !fetchBidsActivityAllowed {
-			// skip the call to a bidder if fetchBids activity is not allowed
-			// do not add this bidder to allowedBidderRequests
+		auctionPermissions := gdprPerms.AuctionActivitiesAllowed(ctx, bidderRequest.BidderCoreName, bidderRequest.BidderName)
+
+		// privacy blocking
+		if rs.isBidderBlockedByPrivacy(req, auctionReq.Activities, auctionPermissions, bidderRequest.BidderCoreName, bidderRequest.BidderName) {
```

Comment:
@SyntaxNode - we have a use case of recording if a bidder request is blocked by gdpr or by activity control so we can differentiate the cause behind it.

so i suggest we return enum or values from `rs.isBidderBlockedByPrivacy` to tell if a bidder request should be blocked, if so the reason behind it

```
blocked :=  rs.isBidderBlockedByPrivacy(...)
if blocked != 0 {
   continue
}

// or we can make blocked return value a pointer, if it is not nil, we consider it is blocked
```